src/middleware/serverMiddleware.ts

```diff
@@ -13,5 +13,10 @@ export default function (context: Context): void {
     !context.store.state.servers.serverList.length
   ) {
     return context.redirect('/server/add');
+  } else if (
+    context.store.state.servers.serverList.length > 1 &&
+    !context.store.state.servers?.serverUsed?.address
```

Comment:
I think this condition doesn't belong here? If no server is selected you should be redirected to the selection page I'd guess?

Reply:
@ferferga 
Ok, you are right, the login page, redirect to the /select page when the call fails.

BUT However it does it after trying a call with an empty address (in case of 2+ servers in the list) then calling the same URL as Vue (obviously without results).

To avoid this waste of time in the login page (since I know that in this context an empty call is tried), since the Middleware server was already implemented in the login, I thought to add this case and force the redirect to / select.

This translates into less time spent waiting, as it saves a call that definitely fails (when you have 2+ servers listed and none selected).

![image](https://user-images.githubusercontent.com/61033694/150639742-c3d4b218-8b19-4516-99fb-4363ee7b62e1.png)

In the photo, we see that the login took us to /server/select (without the middleWare), but after failing a call (which surely would have failed, with no URL set)

-------------

These are the reasons that led me to take care of this case in the Middleware server.

Reply:
@doc-code-hub Sorry I completely forgot this PR was still unmerged 😅.

I forgot about the redirect in the login page, so what you say makes sense. I hope you also tested the cases where no env var was provided and there are no servers, right? (I assume so as tests should also pick those)

I dwon't put more impediments to the middleware stuff as it's a huge mess right now and I already have it refactored in a local branch after the Pinia migration.

Reply:
Hello @ferferga!
Don't worry, the commits don't run away and neither does my fork :)

Yes, I had personally tested the 3 possible cases:
- No Env Var provided
- Env Var with 1 Server
- Env Var with 2+ Server

I also modified the tests to consider all the cases listed (if my memory serves me right).

Later I go back to the branch and I make sure, as it's been a while now :)

I'll let you know, _but it should be a yes._

---

**Instead** with 

> "I dwon't put more impediments to the middleware stuff as it's a huge mess right now and I already have it refactored in a local branch after the Pinia migration"

 are you asking me to remove this check that I added on the middleware or were you talking in general?

Reply:
@doc-code-hub In general 😄